packages/rich-text/src/test/to-dom.js

```diff
@@ -59,6 +59,42 @@ describe( 'applyValue', () => {
 			movedCount: 0,
 			description: 'should not modify',
 		},
+		{
+			current: '<span data-1="">b</span>',
+			future: '<span>b</span>',
+			movedCount: 0,
+			description: 'should remove attribute',
+		},
+		{
+			current: '<span data-1="" data-2="">b</span>',
+			future: '<span>b</span>',
```

Comment:
In case anyone was wondering, this test case would fail in `master`. Added a comment: https://github.com/WordPress/gutenberg/pull/15070/files#diff-829f9f4e942450ee8c33f082c016b173R212.